### PR TITLE
Get values from type fields

### PR DESCRIPTION
**Pull request by Beacze**
Use TypeInterface to define common methods for type enums.

### CODE REVIEW

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    /**
+     * Get size of related objects returned from ElasticSearch index.
+     *
+     * @param object
+     *            JSONObject
+     * @return size of array with related objects
+     */
+    default int getSizeOfProperty(JsonObject object) throws DataException {
+        if (object != null) {
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java]: Why did you add here a `null` check but not in the other methods which access the given `JsonObject` in the same way? Maybe you can switch to `Objects.nonNull` call.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+     * Get boolean value from given json object.
+     * 
+     * @param jsonObject
+     *            returned from ElasticSearch index
+     * @return boolean value for given json
+     */
+    default boolean getBooleanValue(JsonObject jsonObject) throws DataException {
+        try {
+            return jsonObject.getBoolean(this.toString());
+        } catch (RuntimeException e) {
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java]: Why did you catch any `RuntimeException` based class here? So far as I can see only `NullPointerException ` and `ClassCastException` are official thrown.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+     * Get int value from given json object.
+     * 
+     * @param jsonObject
+     *            returned from ElasticSearch index
+     * @return int value for given json
+     */
+    default int getIntValue(JsonObject jsonObject) throws DataException {
+        try {
+            return jsonObject.getInt(this.toString());
+        } catch (RuntimeException e) {
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java]: Why did you catch any `RuntimeException` based class here? So far as I can see only `NullPointerException ` and `ClassCastException` are official thrown.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+     * Get String value from given json object.
+     * 
+     * @param jsonObject
+     *            returned from ElasticSearch index
+     * @return String value for given json
+     */
+    default String getStringValue(JsonObject jsonObject) throws DataException {
+        try {
+            return jsonObject.getString(this.toString());
+        } catch (RuntimeException e) {
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java]: Why did you catch any `RuntimeException` based class here? So far as I can see only `NullPointerException ` and `ClassCastException` are official thrown.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+     * Get json array value from given json object.
+     *
+     * @param jsonObject
+     *            returned from ElasticSearch index
+     * @return json array for given json
+     */
+    default JsonArray getJsonArray(JsonObject jsonObject) throws DataException {
+        try {
+            return jsonObject.getJsonArray(this.toString());
+        } catch (RuntimeException e) {
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java]: Why did you catch any `RuntimeException` based class here? So far as I can see only `ClassCastException` is official thrown.

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+     * @param object
+     *            JSONObject
+     * @return size of array with related objects
+     */
+    default int getSizeOfProperty(JsonObject object) throws DataException {
+        if (object != null) {
+            try {
+                JsonArray jsonArray = (JsonArray) object.get(this.toString());
+                return jsonArray.size();
+            } catch (RuntimeException e) {
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java]: Why did you catch any `RuntimeException` based class here?

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java (hunk 2 above)
Comment (Beacze) [reply Kitodo-DataManagement/src/main/java/org/kitodo/data/elasticsearch/index/type/enums/TypeInterface.java]: Aren't they both RuntimeExceptions?
